iam.tf

```diff
@@ -122,11 +122,14 @@ resource "aws_iam_role" "autoscaler" {
     ]
   })
 
-  inline_policy {
-    name   = "ec2-autoscaler-${var.worker_pool_id}"
-    policy = data.aws_iam_policy_document.autoscaler[count.index].json
-  }
-
   depends_on = [module.asg]
   tags       = var.additional_tags
 }
+
+resource "aws_iam_role_policy" "autoscaler" {
```

Comment:
Pretty sure this change is actually breaking for anyone that has an earlier version deployed. The user will need to write an `import` block in order to make it non-breaking. See the discussion on the provider repo: https://github.com/hashicorp/terraform-provider-aws/issues/39376#issuecomment-2363922272

Reply:
Just to mention as discussed in https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/pull/103#issuecomment-2465023369, it looks like this doesn't create a backwards-compatibility issue. This PR just needs to be rebased against `main` now.

Reply:
Rebased :)
(sorry, don't check GitHub too often so I missed this)

Reply:
Thanks a lot @jeohist.

Sorry to be a pain, but could I ask you to do one more thing - update the version number [here](https://github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2/blob/5b0e2051294b532bd5c58951c39f489ed179e09e/.spacelift/config.yml#L2) to `2.6.4`? It just means we can get the module tests to run to sanity check the changes look good.

Reply:
Sure, done!